### DISCUSSION TITLE
Best practice for package extensions with Makie

### DISCUSSION THREAD

**Discussion by moyner** [JuliaPlots/Makie.jl#2784] [Q&A] [answered]
Hi,

The [package extensions](https://pkgdocs.julialang.org/dev/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)) are coming to Julia 1.9 and I have a plotting package that uses Makie that I'd like to turn into an extension for my main package so that if a user has Makie available then plotting functions become functional. Currently my plotting is made up of two parts:

- Basic plots (line plots, bar plots, ...) that work with both CairoMakie and GLMakie.
- Advanced 3D plots that include some interactables that only work in GLMakie.

The backends re-export Makie but can I just depend Makie? Or is GLMakie the right choice? Does anyone have any experience with Makie as a conditional dependency? Anything I should avoid?

**Comment by asinghvi17** [comment, marked as answer]
Interactive elements are usually defined in a backend-independent way, so just depending on Makie should be fine.  

I just did something like this for Rasters.jl (the PR is still up) but it's a bit kludgey for now, since figure-level recipes aren't really a thing in Makie yet.
